srv/components/provisioner/salt_master/config.sls

```diff
@@ -20,6 +20,22 @@ salt_master_config_updated:
     - name: /etc/salt/master
     - source: salt://components/provisioner/salt_master/files/master
 
+# Always start glusterfshsaredstorage before salt-master
+Update glusterfssharedstorage.service:
+  file.managed:
+    - name: /usr/lib/systemd/system/glusterfssharedstorage.service
+    - source: salt://components/provisioner/files/glusterfshsaredstorage.service
+
+Update salt-master.service:
+  file.managed:
+    - name: /usr/lib/systemd/system/salt-master.service
+    - source: salt://components/provisioner/salt_master/files/salt-master.service
+
+Reload updated services:
+  cmd.run:
+    - name: systemctl daemon-reload
+    - onchanges:
+      - file: Update salt-master.service
```

Comment:
I believe the same relation is required for the last state to restart salt-master in the background since in general it may impact the way the service is started.